pkg/cmd/tasks/initcmd/init.go

```diff
@@ -36,50 +37,75 @@ func New(c *cli.Config) *cobra.Command {
 		},
 	}
 
-	cmd.Flags().StringVarP(&cfg.file, "file", "f", "airplane.yml", "Path to a file to store task definition")
+	cmd.Flags().StringVarP(&cfg.file, "file", "f", "", "Path to a file to store task definition")
 	cmd.Flags().StringVar(&cfg.from, "from", "", "Slug of an existing task to generate from")
 
-	// --from is the only way to use `init` for now. We'll soon add
-	// a way to be prompted through the creation of a task def.
-	cli.Must(cmd.MarkFlagRequired("from"))
-
 	return cmd
 }
 
 func run(ctx context.Context, cmd *cobra.Command, cfg config) error {
-	var client = cfg.cli.Client
+	var kind initKind
+	var err error
+	// If --from is provided, we already know the user wants to create
+	// from an existing task, so we don't need to prompt the user here.
+	if cfg.from == "" {
+		cmd.Printf("Airplane is the code-first solution for engineers building internal tools.\n\n")
+		cmd.Printf("This command will configure a task definition which specifies how to deploy your task to Airplane.\n\n")
 
-	res, err := client.GetTask(ctx, cfg.from)
-	if err != nil {
-		return errors.Wrap(err, "get task")
+		if kind, err = pickInitKind(); err != nil {
+			return err
+		}
+	} else {
+		kind = initKindTask
 	}
 
-	dir, err := taskdir.Open(cfg.file)
-	if err != nil {
-		return errors.Wrap(err, "opening task directory")
+	switch kind {
+	case initKindSample:
+		if err := initFromSample(cmd, cfg); err != nil {
+			return err
+		}
+	case initKindScratch:
+		if err := initFromScratch(cmd, cfg); err != nil {
+			return err
+		}
+	case initKindTask:
+		if err := initFromTask(ctx, cmd, cfg); err != nil {
+			return err
+		}
+	default:
+		return errors.Errorf("Unexpected unknown initKind choice: %s", kind)
 	}
-	defer dir.Close()
 
-	if err := dir.WriteDefinition(taskdir.Definition{
-		Slug:           res.Slug,
-		Name:           res.Name,
-		Description:    res.Description,
-		Image:          res.Image,
-		Command:        res.Command,
-		Arguments:      res.Arguments,
-		Parameters:     res.Parameters,
-		Constraints:    res.Constraints,
-		Env:            res.Env,
-		ResourceLimits: res.ResourceLimits,
-		Builder:        res.Builder,
-		BuilderConfig:  res.BuilderConfig,
-		Repo:           res.Repo,
-		Timeout:        res.Timeout,
-	}); err != nil {
-		return errors.Wrap(err, "writing task definition")
-	}
+	return nil
+}
 
-	cmd.Printf("Created an Airplane task definition for %s in %s\n", res.Name, cfg.file)
+type initKind string
 
-	return nil
+var (
+	initKindSample  initKind = "Create from an Airplane-provided sample"
+	initKindScratch initKind = "Create from scratch"
+	initKindTask    initKind = "Create from an existing Airplane task"
```

Comment:
Maybe we should make this read more like an import? It's not quite the same as the others...

```suggestion
	initKindTask    initKind = "Import definition from an existing Airplane task"
```

Reply:
Went with "clone"!

Reply:
```sh
❯ lairlocal tasks init --from echo_loop
X Sorry, your reply was invalid: You must pick a new slug.
? Pick a unique identifier (slug) for this task echo_loop_2

An Airplane task definition for 'Echo Loop' has been created!

To deploy it to Airplane, run:
  airplane tasks deploy -f airplane.yml

❯ cat airplane.yml
slug: echo_loop_2
# ...
```